### DISCUSSION TITLE
Support Alert Plugin

### DISCUSSION THREAD

**Discussion by strahe** [filecoin-project/curio#83] [Ideas]
Hi, I have a  suggestion regarding the `alertmanager` that I hope we can discuss: 
Currently, the `alertmanager` uses the PageDuty service, which has quite a few limitations for companies that already have a notification service. 
Could we consider allowing the `AlertTask` to support plugins?  For example: 

Define a plugin interface:
```go
type AlertPlugin interface {
    SendAlert(data *pdPayload) error
}
```
And the AlertTask structure:
```go
type AlertTask struct {
    alertPlugin AlertPlugin
}
```
Then, in the creation of a new AlertTask:
```go
func NewAlertTask(full api.FullNode, db *harmonydb.DB, cfg config.AlertingConfig) *AlertTask {
    var plugin AlertPlugin
    // Assuming there's a function loadAlertPlugin, which loads a plugin based on the configuration
    plugin = loadAlertPlugin(cfg)
    return &AlertTask{
        // Initialize other fields...
        alertPlugin: plugin,
    }
}
```
And in the sendAlert function:

```go
func (a *AlertTask) sendAlert(data *pdPayload) error {
    return a.alertPlugin.SendAlert(data)
}
```

This way, we can define and embed a variety of options in the future.

**Comment by LexLuthr** [comment]
Sounds like a good idea. We are happy to get any feedback and code from community to improve alerting. Do you want to write up the PR?

**Reply by strahe** [reply]
Sure, I'll prepare the PR for the alerting improvement and submit it for review.

**Reply by LexLuthr** [reply]
Great! Looking forward to your PR. Thank you for helping improve Curio.

**Comment by LexLuthr** [comment]
Closing as implemented in https://github.com/filecoin-project/curio/pull/89
